src/robot/running/keywordrunner.py

```diff
@@ -113,7 +114,12 @@ def _get_and_report_failure(self):
         return failure
 
 
-class ForLoopRunner(object):
+class BasicForLoopRunner(object):
+    """A basic For Loop.
+
+    To make a new kind of For loop, subclass this and override
+    _flavor_name() and _transform_items(items).
+    """
```

Comment:
I don't think this doc provides much added value. If we wont' to make it explicit how to extend, it would be better to have `ForRunnerBase` with the abstract methods raising `NotImplementedError` or actually marked abstract using functionality in the `abc` module.